src/libsyntax/config.rs

```diff
@@ -299,6 +350,13 @@ impl<'a> fold::Folder for StripUnconfigured<'a> {
         // Interpolated AST will get configured once the surrounding tokens are parsed.
         mac
     }
+
+    fn fold_pat(&mut self, pattern: P<ast::Pat>) -> P<ast::Pat> {
+        let mut pattern = pattern.unwrap();
+        pattern.node = self.configure_pat_kind(pattern.node);
+        println!("{:#?}", pattern);
+        fold::noop_fold_pat(P(pattern), self)
```

Comment:
It's more efficient and idiomatic to do the following here:
```rust
pattern.map(|mut pattern| {
    // ...
    pattern
})
```
(if you did `configure_pat` instead of `configure_pat_kind`, the `.map` would be there)